File: Napakalaki/src/napakalaki/NumericBadConsequence.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki;

import java.util.ArrayList;

/**
 *
 * @author chen__000
 */
public class NumericBadConsequence extends BadConsequence {
    private int nVisibleTreasures;
    private int nHiddenTreasures;
    
        public NumericBadConsequence(String txt, int lvs, int nVsbl, int nHddn){
        super(txt,lvs);
        nVisibleTreasures=nVsbl;
        nHiddenTreasures=nHddn;
        }
        
        
        public int getnVisibleTreasures(){
           return nVisibleTreasures;
         }
        
         public int getnHiddenTreasures(){
            return nHiddenTreasures;
         }
        
        @Override
        public boolean isEmpty(){
           boolean empty=false;
           if(levels==0&&nHiddenTreasures==0&&nVisibleTreasures==0){
           empty=true;
           }
          return empty;
        }
        
    @Override
        public void substractVisibleTreasure(Treasure t){
        if (this.nVisibleTreasures != 0){
            nVisibleTreasures--;
        }
    }
        
    @Override
        public void substractHiddenTreasure(Treasure t){
        if (this.nHiddenTreasures != 0){
            nHiddenTreasures--;
        }
    }
        
    @Override
 public NumericBadConsequence adjustToFitTreasureLists(ArrayList<Treasure> v,ArrayList<Treasure> h){
     
        
        NumericBadConsequence resultado;
        levels=0;
        
            int visibles,ocultos;
            
            visibles=Math.min(v.size(), this.nVisibleTreasures);
            ocultos=Math.min(h.size(), this.nHiddenTreasures);
            
            resultado=new NumericBadConsequence(this.getText(),this.getLevels(),visibles,ocultos);
        
        
        return resultado;
    }

}

File: Napakalakiruby/lib/numeric_bad_consequence.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative "bad_consequence.rb"

class NumericBadConsequence<BadConsequence
  attr_accessor:nVisibleTreasures,:nHiddenTreasures
  
  def initialize(txt,lv,nv,nh)
    @text=txt
    @levels=lv
    @nVisibleTreasures=nv
    @nHiddenTreasures=nh
  end
  
  def isEmpty
    empty=false
      if(@levels==0&&@nHiddenTreasures==0&&@nVisibleTreasures==0)
        empty=true
      end
      return empty
  end
  
      def substractVisibleTreasure(t)
      if(@nVisibleTreasures>0)
        @nVisibleTreasures=@nVisibleTreasures-1
      end
    end
    
    def substractHiddenTreasure(t)
      if(@nHiddenTreasures>0)
        @nHiddenTreasures=@nHiddenTreasures-1
      end
    end
end
